RA_Integration/RA_Dlg_Memory.cpp

```diff
@@ -1205,6 +1233,42 @@ void Dlg_Memory::OnLoad_NewRom()
 	else
 		SetDlgItemText( g_MemoryDialog.m_hWnd, IDC_RA_WATCHING, L"Loading..." );
 
+	if( g_MemManager.TotalBankSize() > 0 )
+	{ 
+		ByteAddress start, end;
+		if( GetSystemMemoryRange( start, end ) )
+		{
+			wchar_t label[64];
+			if( g_MemManager.TotalBankSize() > 0x10000 )
+				wsprintf( label, L"System Memory (0x%06X-0x%06X)", start, end );
+			else
+				wsprintf( label, L"System Memory (0x%04X-0x%04X)", start, end );
+
+			SetDlgItemText( g_MemoryDialog.m_hWnd, IDC_RA_CBO_SEARCHSYSTEMRAM, label );
+		}
+		else
+		{
+			SetDlgItemText( g_MemoryDialog.m_hWnd, IDC_RA_CBO_SEARCHSYSTEMRAM, L"System Memory (unspecified)" );
+			EnableWindow( GetDlgItem( g_MemoryDialog.m_hWnd, IDC_RA_CBO_SEARCHSYSTEMRAM ), FALSE);
+		}
+
+		if( GetGameMemoryRange( start, end ) )
+		{
+			wchar_t label[64];
+			if( g_MemManager.TotalBankSize() > 0x10000 )
+				wsprintf( label, L"Game Memory (0x%06X-0x%06X)", start, end );
+			else
+				wsprintf( label, L"Game Memory (0x%04X-0x%04X)", start, end );
+
+			SetDlgItemText( g_MemoryDialog.m_hWnd, IDC_RA_CBO_SEARCHGAMERAM, label );
+		}
+		else
+		{
+			SetDlgItemText( g_MemoryDialog.m_hWnd, IDC_RA_CBO_SEARCHGAMERAM, L"Game Memory (unspecified)" );
+			EnableWindow( GetDlgItem( g_MemoryDialog.m_hWnd, IDC_RA_CBO_SEARCHGAMERAM ), FALSE);
+		}
+	}
```

Comment:
This block initializes the System and Game memory range labels once a ROM has been loaded and the system is known.